Multisig/UI/Assets/AssetsViewModel.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  AssetsViewModel.swift
+//  Multisig
+//
+//  Created by Andrey Scherbovich on 13.05.20.
+//  Copyright © 2020 Gnosis Ltd. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+struct TokenBalance: Identifiable {
+    var id: String {
+        return address
+    }
+    var imageURL: URL {
+        return URL(string: "https://gnosis-safe-token-logos.s3.amazonaws.com/\(address).png")!
+    }
+    let address: String
+    let balance: String
+    let balanceUsd: String
+}
+
+class AssetsViewModel: ObservableObject {
+    @Published var balances = [TokenBalance]()
+    @Published var isLoading: Bool = true
+    @Published var errorMessage: String? = nil
+
+    private var subscribers = Set<AnyCancellable>()
+
+    init() {
+        let request = Safe.fetchRequest().selected()
+        let selectedSafe = try! CoreDataStack.shared.viewContext.fetch(request).first!
+        isLoading = true
+        Just(selectedSafe.address!)
```

Comment:
this is not good, I'd rather `Just(address).compactMap { $0 }` - we don't have a requirement that the address is going to be known at all times, and this model will be more useful.

Reply:
As the safe is passed in the initializer, the address will always be there.